### ISSUE TITLE
Example given to show that it is impossible to go from a bitcoin address to a public key is not clear

### ISSUE THREAD

**Issue by magevadi** [ProgrammingBlockchain/ProgrammingBlockchain#77]
At this page:
https://programmingblockchain.gitbooks.io/programmingblockchain/content/bitcoin_transfer/private_key.html

In this example:
PubKey publicKey = privateKey.PubKey;
BitcoinPubKeyAddress bitcoinPublicKey = publicKey.GetAddress(Network.Main); // 1PUYsjwfNmX64wS368ZR5FMouTtUmvtmTY
//PubKey samePublicKey = bitcoinPublicKey.ItIsNotPossibl

Issues are:
1. Bitcoin address is shown to be generated from public key instead of public key hash. This is not consistent with previous examples. Previous examples show that the bitcoin address is created from Public Key Hash instead of public key itself. 
2. There is no true/false assert statement style code. Having a boolean assertion explains things more clearly.

**Comment by NicolasDorier** [comment]
I agree, can you make a PR for this? `pubkey.Hash.GetAddress(Network.Main)` might be more coherent.

**Comment by nopara73** [comment]
ACK
